src/main/java/com/bobocode/Dialect.java

```diff
@@ -1,5 +1,26 @@
 package com.bobocode;
 
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public interface Dialect {
+    String SELECT_ALL_TEMPLATE = "select * from %s";
+
+    String SELECT_ALL_ID_TEMPLATE = "select * from %s where id = ?";
+
+    String SELECT_ALL_BY_PROPERTIES_TEMPLATE = "select * from %s where %s";
+
+    String AND_WITH_SPACES = " AND ";
+
+    String NAME_EQUALS_VALUE_TEMPLATE = "%s = ?";
+
+    static String prepareWhereClause(Map<String, Object> properties) {
+        return properties.keySet().stream()
+                .sorted()
+                .map(NAME_EQUALS_VALUE_TEMPLATE::formatted)
+                .collect(Collectors.joining(AND_WITH_SPACES));
+    }
+
+    String getLimitOffsetClause(Long limit, Long offset);
```

Comment:
@DanielChanger what to do if user passes limit = 0? should we load the whole table?
I guess we have to prevent it and to add checking. What do you think?

Reply:
@DanielChanger what to do if user passes limit = 0? should we load the whole table?
I guess we have to prevent it and to add checking. What do you think?

Reply:
@DanielChanger Roman and I discussed this question and he noticed user might want to load the whole table if it is some config data. So the best solution is to add one more method without limit and offset arguments (like it was at the beginning :) ). At the same time I'll add the check that limit and offset  > 0.

Reply:
Okay, make sense to me :)